src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -1728,9 +1728,28 @@ def q_impl(
             *q_inputs, calibrate=False, quantize_actual_values=True
         )
 
-        # Retrieve values and axes parameters
+        assert_true(
+            isinstance(prepared_inputs[0], (QuantizedArray)),
+            "Prepared inputs's values should be a QuantizedArray. "
+            f"Got {type(prepared_inputs[0])}",
+        )
+
+        assert_true(
+            isinstance(prepared_inputs[1], numpy.ndarray),
+            "ReduceSum axis parameter should either be a Numpy array or None. "
+            f"Got {type(prepared_inputs[1])}",
+        )
+
+        # Retrieve the quantized values and the axes to sum over
+        # Parameter "axis" Numpy's sum operator has to be a tuple (and not an array)
         q_values = prepared_inputs[0].qvalues
-        axes = prepared_inputs[1]
+        axes = tuple(prepared_inputs[1])
+
+        assert_true(
+            0 not in axes,
```

Comment:
@jfrery @fd0r here I make sure we do not sum over axis 0 in any case

Reply:
Yes I think that's correct in our current settings.